### PR TITLE
Discovery

### PR DESCRIPTION
**Pull request by AaronDavidSchneider**
Extension of fritzbox tools to use the discovery component. 

I tested (as appropriate):
- [x] Yaml Mode

Closes issue: #86 
Related issue: #41

### CODE REVIEW

**Comment by AaronDavidSchneider** [comment]
Seems to work. However, YAML mode seems to be broken. Need to find a fix for that.

**Comment by AaronDavidSchneider** [comment]
YAML error fixed. Ready for review 👯‍♂️

**Comment by AaronDavidSchneider** [comment]
Small addition: Take port from discovery.

**Comment by AaronDavidSchneider** [comment]
Discovery (I have two fritzboxes, one is the modem (7530), the other one is just a repeater (mesh client)):
Screenshot 2020-10-01 at 13 33 51

When clicking on configure:
Screenshot 2020-10-01 at 13 34 08

Once one fritzbox is added the other fritzbox will not get discovered anymore (only one component allowed) once hass is restarted.

File: custom_components/fritzbox_tools/config_flow.py [hunk 1]
```diff
@@ -41,12 +47,59 @@ class FritzBoxToolsFlowHandler(ConfigFlow):
...
+        # abort if already configured
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_configured")
+
+        self.context["title_placeholders"] = {"name": self._name.replace("FRITZ!Box ","")}
+        return await self._show_setup_form_confirm()
+
+    async def async_step_confirm(self, user_input=None):
+        """Handle user-confirmation of discovered node."""
+        errors = {}
```
Comment (mammuth) [inline custom_components/fritzbox_tools/config_flow.py:75]: Duplicated declaration

**Review by mammuth — CHANGES_REQUESTED** [review]

**Comment by mammuth** [comment]
Pretty cool! 🚀  Only left a small comment.

**Comment by AaronDavidSchneider** [comment]
Thanks for the review @mammuth. I removed the duplicate variable `errors`

**Comment by AaronDavidSchneider** [comment]
Shall we merge this PR @mammuth?

**Comment by mammuth** [comment]
Yes. It's no breaking change that should be announced in the release notes, right?

**Comment by AaronDavidSchneider** [comment]
> Yes. It's no breaking change that should be announced in the release notes, right?

No. Thanks!

**Pull request merged**